nav2_costmap_2d/src/layered_costmap.cpp

```diff
@@ -145,40 +145,46 @@ void LayeredCostmap::updateMap(double robot_x, double robot_y, double robot_yaw)
   minx_ = miny_ = std::numeric_limits<double>::max();
   maxx_ = maxy_ = std::numeric_limits<double>::lowest();
 
-  for (vector<std::shared_ptr<Layer>>::iterator plugin = plugins_.begin();
-    plugin != plugins_.end(); ++plugin)
+  #pragma omp parallel reduction(min:minx_) reduction(min: miny_) \
+  reduction(max: maxx_) reduction(max: maxy_) num_threads(2)
```

Comment:
Is there a way to have it autoscale the number of threads by compute available resources or does it need to be specified?

Reply:
The margin of improvement (at least with default number layers) is soo small that launching extra threads becomes more costly than doing it in a serial way. A more reasonable approach would be to check how many (plugins + filters) there are present at runtime and set the number by that.

Reply:
Do the openMP preprocessor statements allow you to do that? I would think it would be set at compile time. Though if you had many plugins (6) on a normal machine with only 4 cores, it would necessarily help. I think at most it should be the number of cores on the computer (which to my understanding of OpenMP is actually the default if not specified) but less if less plugins or some of them are very light weight.

Reply:
Wasn't sure about the runtime decision on a number of threads, so i've tried this demo and it actually gives a number of threads requested at runtime.

```
  auto a = random_int();
  
  std::cout << a << '\n';

#pragma omp parallel num_threads(a) default(none)
  {
    int b = omp_get_num_threads();
	printf("%d", b);
  }
```
So let's set that number to be something along the lines:

```
int n_threads = std::min(number_of_cores, (number_of_plugins + number_of_filters))

Reply:
Aren't the plugins / filters in different `for` loops? I don't think we'd need to sum them unless the parallelization can continue to the send batch of the filters

Reply:
Yeah, we could have a separate number for plugins & filters